internal/authentication/tokengetter.go

```diff
@@ -19,6 +21,15 @@ type TokenGetter struct {
 	mu                 sync.RWMutex
 }
 
+type ServiceAccountNotFoundError struct {
+	ServiceAccountName string // The name of the missing ServiceAccount.
+}
```

Comment:
Maybe not strictly necessary, but it may prove useful to:
1. include the underlying error value in the struct
2. implement the Unwrap function for this struct so that the underlying error can be recovered if necessary.

I recognize that we don't want to show up in the error string, so I'm _not_ suggesting that we use `fmt.Errorf("blah blah: %w", err)`

But we can do this:
```go
type ServiceAccountNotFoundError struct {
	ServiceAccountName string // The name of the missing ServiceAccount.
	Err error // The underlying error
 }

func (e *ServiceAccountNotFoundError) Unwrap() error {
	return e.Err
}
```